pkg/services/sqlstore/tls_mysql.go

```diff
@@ -16,18 +19,16 @@ func makeCert(tlsPoolName string, config DatabaseConfig) (*tls.Config, error) {
 	if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
 		return nil, err
 	}
-	clientCert := make([]tls.Certificate, 0, 1)
-	if config.ClientCertPath != "" && config.ClientKeyPath != "" {
 
-		certs, err := tls.LoadX509KeyPair(config.ClientCertPath, config.ClientKeyPath)
-		if err != nil {
-			return nil, err
-		}
-		clientCert = append(clientCert, certs)
-	}
 	tlsConfig := &tls.Config{
-		RootCAs:      rootCertPool,
-		Certificates: clientCert,
+		RootCAs: rootCertPool,
+	}
+	if config.ClientCertPath != "" && config.ClientKeyPath != "" {
+		tlsConfig.GetClientCertificate = func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
+			tlslog.Debug("Loading client certificate")
+			cert, err := tls.LoadX509KeyPair(config.ClientCertPath, config.ClientKeyPath)
```

Comment:
So how often will this be called? Very hard to understand since I cannot find any reference to `GetClientCertificate` in go mysql source code.

Reply:
This func will be called each time a new mysql connection is opened (potentially more if renegotiation happens). It's part of the [tls](https://golang.org/src/crypto/tls/common.go) package:
```
// GetClientCertificate, if not nil, is called when a server requests a
// certificate from a client. If set, the contents of Certificates will
// be ignored.
//
// If GetClientCertificate returns an error, the handshake will be
// aborted and that error will be returned. Otherwise
// GetClientCertificate must return a non-nil Certificate. If
// Certificate.Certificate is empty then no certificate will be sent to
// the server. If this is unacceptable to the server then it may abort
// the handshake.
//
// GetClientCertificate may be called multiple times for the same
// connection if renegotiation occurs or if TLS 1.3 is in use.
```

Reply:
Not sure it's a good idea to read cert files on every single mysql connection.

Reply:
I believe we're using a connection pool for mysql, so the impact really depends on the `max_open_conn` and `conn_max_lifetime` settings. Is there an alternative approach that you prefer? The article you linked to was doing timed rotations similar to my original implementation.

Reply:
An alternative approach that would satisfy my requirements is to add an endpoint to trigger a reload of the client certs

Reply:
@marefr ^ any thoughts here?